### ISSUE TITLE
[@typescript-eslint/no-use-before-define] Unexpected warning in terminal

### ISSUE THREAD

**Issue by lindomar-oliveira** [typescript-eslint/typescript-eslint#2533]
I did as described in this comment https://github.com/typescript-eslint/typescript-eslint/issues/2502#issuecomment-689595020, but the following warning: `'React' was used before it was defined` is still there. However, when I execute the linter command, nothing is displayed in the terminal.

##### eslintrc.json

```json
  ...
  "rules": {
    "no-use-before-define": "off",
    "@typescript-eslint/no-use-before-define": ["error"]
  }
  ...
}
```
**Versions**

| package                                       | version |
| ------------------------------------------- | ---------- |
| `@typescript-eslint/eslint-plugin` | `4.1.0`   |
| `@typescript-eslint/parser`          | `4.1.0`   |
| `TypeScript`                                 | `4.0.2`   |
| `ESLint`                                       | `6.6.0`   |
| `node`                                          | `12.8.3` |

**Screenshots**

[1 - Warnings in `npm start`](https://i.imgur.com/mU06gNJ.png)
[2 - Zero warnings in `npm lint`](https://i.imgur.com/Q2yqfTv.png)

**Comment by wsfuller** [comment]
I did the following as above and was tempted to put up a new issue since the previous has been closed and the answers in the thread did not help resolve the problem. Followed the steps in #2502 and still seeing the issue.

`./src/index.tsx
  Line 1:8:  'React' was used before it was defined  @typescript-eslint/no-use-before-define`

I'm trying to get my first React TypeScript project up and running and unable to get the this linting correctly out of the box.

## Repro steps:
- `npx create-react-app my-app --template typescript`
- `npm install install --save-dev @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint-config-airbnb-typescript@6.3.2 eslint-plugin-jest eslint-plugin-import eslint-plugin-jsx-a11y eslint-plugin-react-hooks eslint-plugin-react prettier eslint-config-prettier eslint-plugin-prettier

```
.eslintrc.js
...
rules: {
  "no-use-before-define": "off",
  "@typescript-eslint/no-use-before-define": ["error"] 
}
...
```

I also tried the suggestion by [@linonetwo](https://github.com/typescript-eslint/typescript-eslint/issues/2502#issuecomment-689595020) and that suggestion doesn't help either.

```
...
rules: {
  'no-use-before-define': [0],
  '@typescript-eslint/no-use-before-define': [1],
}
...
```

**Comment by bradzacher** [comment]
[`create-react-app` is currently restricted to v3 of our tooling.](https://github.com/facebook/create-react-app/blob/c8ea284ab38cd57d3fdffa16d21702fd02c4623b/packages/react-scripts/package.json#L34-L35)

you can consider using something like yarn resolutions to force the upgrade, otherwise you need to wait for them to upgrade.
